app/services/liftigniter.js

```diff
@@ -0,0 +1,112 @@
+import Ember from 'ember';
+import localStorageConnector from '../utils/local-storage-connector';
+
+const {Service, RSVP, inject} = Ember;
+
+export default Service.extend({
+	fastboot: inject.service(),
+
+	initLiftigniter(adsContext) {
+		if (this.get('fastboot.isFastBoot') || !window.liftigniter) {
+			return;
+		}
+
+		const kxallsegs = localStorageConnector.getItem('kxallsegs');
+		let context = {};
+
+		if (adsContext && adsContext.targeting) {
+			const targeting = adsContext.targeting;
+
+			context = {
+				_wCategory: targeting.wikiCategory,
+				_wName: targeting.wikiDbName,
+				_wTop: targeting.wikiIsTop1000,
+				_wikiLanguage: targeting.wikiLanguage,
+				_wVert: targeting.wikiVertical,
+				_pType: targeting.pageType,
+				_pName: targeting.pageName
+			};
+
+			if (targeting.wikiCustomKeyValues) {
+				targeting.wikiCustomKeyValues
+					.split(';')
+					.map((keyVal) => {
+						return keyVal.split('=');
+					})
+					.forEach((parts) => {
+						const key = `_${parts[0]}`;
+
+						if (!context[key]) {
+							context[key] = [parts[1]];
+						} else {
+							context[key].push(parts[1]);
+						}
+					});
+			}
+		}
+
+		if (kxallsegs) {
+			context._kruxTags = kxallsegs.split(',');
+		}
+
+		window.liftigniter('init', 'l9ehhrb6mtv75bp2', {
+			config: {
+				sdk: {
+					queryServer: '//query.fandommetrics.com'
+				},
+				activity: {
+					activityServer: '//api.fandommetrics.com'
+				},
+				inventory: {
+					inventoryServer: '//api.fandommetrics.com'
+				},
+				globalCtx: context,
+			}
+		});
+
+		window.liftigniter('send', 'pageview');
+		window.liftigniter('setRequestFields', ['rank', 'thumbnail', 'title', 'url', 'presented_by', 'author']);
+	},
+
+	getData(config) {
+		const deferred = RSVP.defer(),
```

Comment:
more common way of ding this is
return RSVP.Promise((resolve, reject) {
...
resolve(response)
})

but I would not change it now,